File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/BlocksResource.java
package work.socialhub.kmastodon.api;

import mastodon4j.Range;
import mastodon4j.entity.Account;
import mastodon4j.entity.share.Response;

/**
 * @author hecateball
 */
public interface BlocksResource {

    /**
     * Fetching a user's blocks.
     *
     * @return an array of Accounts blocked by the authenticated user
     */
    public Response<Account[]> getBlocks();

    /**
     * Fetching a user's blocks.
     *
     * @param range
     * @return an array of Accounts blocked by the authenticated user
     */
    public Response<Account[]> getBlocks(Range range);
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/api/BlocksResource.kt
package work.socialhub.kmastodon.api

import work.socialhub.kmastodon.api.request.blocks.BlocksBlocksRequest
import work.socialhub.kmastodon.api.response.Response
import work.socialhub.kmastodon.api.response.blocks.BlocksBlocksResponse
import kotlin.js.JsExport


@JsExport
interface BlocksResource {

    /**
     * Fetching a user's blocks.
     */
    suspend fun blocks(
        request: BlocksBlocksRequest
    ): Response<Array<BlocksBlocksResponse>>

    @JsExport.Ignore
    fun blocksBlocking(
        request: BlocksBlocksRequest
    ): Response<Array<BlocksBlocksResponse>>
}
